### PR TITLE
Implement UseNLog extension method

### PR DESCRIPTION
**Pull request by bryan5989**
not ready yet, i only made unit test stubs. I need to implement these still. @304NotModified

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,48 @@
...
+[![](https://sonarcloud.io/api/project_badges/measure?project=nlog.extensions.hosting&metric=vulnerabilities)](https://sonarcloud.io/dashboard/?id=nlog.extensions.hosting) 
+[![](https://sonarcloud.io/api/project_badges/measure?project=nlog.extensions.hosting&metric=code_smells)](https://sonarcloud.io/project/issues?id=nlog.extensions.hosting&resolved=false&types=CODE_SMELL) 
+[![](https://sonarcloud.io/api/project_badges/measure?project=nlog.extensions.hosting&metric=duplicated_lines_density)](https://sonarcloud.io/component_measures/domain/Duplications?id=nlog.extensions.hosting) 
+[![](https://sonarcloud.io/api/project_badges/measure?project=nlog.extensions.hosting&metric=sqale_debt_ratio)](https://sonarcloud.io/dashboard/?id=nlog.extensions.hosting) 
+[![](https://sonarcloud.io/api/project_badges/measure?project=nlog.extensions.hosting&metric=coverage)](https://sonarcloud.io/component_measures?id=nlog.extensions.hosting&metric=coverage) 
+
+[NLog](https://github.com/NLog/NLog) provider for [Microsoft.Extensions.Logging](https://github.com/aspnet/Logging); .NET Core.
```
Comment (304NotModified) [inline README.md]: we need to update this one

File: src/NLog.Extensions.Hosting/NLog.Extensions.Hosting.csproj [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <RootNamespace>NLog</RootNamespace>
+    <Company>Nlog</Company>
+    <Authors>Microsoft;Julian Verdurmen;Bryan Gonzalez</Authors>
+    <Description>NLog extension for Microsoft.Extensions.Hosting for usage in .NET Standard libraries and net core applications making use of the generic Host class introduced in netcore 2.1</Description>
+    <PackageLicenseUrl>https://github.com/NLog/NLog.Extensions.Logging/blob/master/LICENSE</PackageLicenseUrl>
```
Comment (304NotModified) [inline src/NLog.Extensions.Hosting/NLog.Extensions.Hosting.csproj]: Log.Extensions.Logging -> Log.Extensions.Hosting

File: src/NLog.Extensions.Hosting/NLog.Extensions.Hosting.csproj [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    <PackageProjectUrl>https://github.com/NLog/NLog.Extensions.Logging</PackageProjectUrl>
```
Comment (304NotModified) [inline src/NLog.Extensions.Hosting/NLog.Extensions.Hosting.csproj]: Log.Extensions.Logging -> Log.Extensions.Hosting

File: src/NLog.Extensions.Hosting/NLog.Extensions.Hosting.csproj [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    <PackageIconUrl>http://nlog-project.org/NConfig.png</PackageIconUrl>
+    <RepositoryUrl>git://github.com/NLog/NLog.Extensions.Logging</RepositoryUrl>
```
Comment (304NotModified) [inline src/NLog.Extensions.Hosting/NLog.Extensions.Hosting.csproj]: Log.Extensions.Logging -> Log.Extensions.Hosting

File: README.md [hunk 5]
```diff
@@ -1 +1,48 @@
...
+- none
+
+
+### How to run the example (aspnet-core-example)
+TODO: this
+How to run the [example](#):
+
+1. Install dotnet: http://dot.net 
+2. From source: `dotnet run`
+3. or, after publish: `dotnet $assemblyname.dll`
```
Comment (304NotModified) [inline README.md]: do we need to update this?

**Comment by bryan5989** [comment]
@304NotModified This is good to go minus the unit tests. Will get to those soon. But I would ask that you take care of the CI stuffs. i don't want to register all the ci stuff to my github. (not a member of any of the orgs)

**Comment by 304NotModified** [comment]
> But I would ask that you take care of the CI stuffs. i don't want to register all the ci stuff to my github. (not a member of any of the orgs)

Yes I will do that :)

I will try to review/test soon, thanks!

**Comment by bryan5989** [comment]
@304NotModified as good as i can make it. Still missing the ci stuff. There's also the examples using a project-project reference since this isn't yet available on Nuget. Additionally, theres an opportunity to clean up the code for the parameter-less UseNLog(). Currently, I'm instantiating a new NLogProviderOptions with an object initializer, with all the properties matching NLogProviderOptions.Default. But it would be nice if I could just use NLogProviderOptions.Default directly, but am unable to due to access level.

**Pull request merged**